### PR TITLE
Feature/interceptor

### PR DESCRIPTION
**Pull request by alettieri**
Introduce HTTP Interceptor. Update gulp test process, removed gulp-karma dependency.

### CODE REVIEW

**Comment by alettieri** [comment]
@prestonvanloon Let me know what you think of these changes. I tried bumping the version to 0.0.1 and accidentally set it to 0.1.0. I merged in your travis-ci and version bump changes.

File: src/interceptors/http.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+(function (angular) {
+  	'use strict';
+	
+	angular.module('newrelic-angular.interceptor.http', []).factory('httpInterceptor', HttpInterceptorFactory)
```
Comment (prestonvanloon) [inline src/interceptors/http.js]: This should be formatted a little better: 

```
angular.module('newrelic-angular.interceptor.http', [])
  .factory('httpInterceptor', HttpInterceptorFactory)
  .config(HttpInterceptorConfig);
```

File: test/interceptors/http.spec.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
+(function () {
+  'use strict';
+
+  describe('$http interceptor ', function () {
+    
+		var $log;
```
Comment (prestonvanloon) [inline test/interceptors/http.spec.js]: Crazy tabbing here! haha

File: test/interceptors/http.spec.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        url: '/testRequest'
+      }
+    };
+    
+    beforeEach(module('newrelic-angular.interceptor.http'));
+    
+    beforeEach(inject(function (_$log_, _$window_, _httpInterceptor_) {
+      $log = _$log_;
+      $window = _$window_;
+			httpInterceptor = _httpInterceptor_;
```
Comment (prestonvanloon) [inline test/interceptors/http.spec.js]: tabbing

**Comment by prestonvanloon** [comment]
Just a few style issues, no big deal. 

Can you bump to 0.1.0? 
That makes more sense to me after merging these changes.

**Comment by alettieri** [comment]
Cool. I'll get those take care of. Wanna throw in a .editorconfig and
lining process? Can take care of that as well

**Comment by prestonvanloon** [comment]
Sounds good to me. Thanks

**Comment by prestonvanloon** [comment]
Looks great to me. Are you ready to merge?

**Comment by alettieri** [comment]
Let's do it.

**Comment by alettieri** [comment]
Dang it. Yes, let's merge.

**Pull request merged**